python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,382 @@
+        """
+        # TODO: Change airflow MssqlHook to fetch database and schema separately
+        database = self.hook.get_connection(self.conn_id).schema
+        return Metadata(database=database, schema=self.DEFAULT_SCHEMA)  # type: ignore
+
+    def schema_exists(self, schema) -> bool:
+        """
+        Checks if a schema exists in the database
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        schema_result = self.hook.run(
+            "SELECT schema_name FROM information_schema.schemata WHERE lower(schema_name) = lower(%(schema_name)s);",
+            parameters={"schema_name": schema.lower()},
+            handler=lambda x: [y[0] for y in x.fetchall()],
+        )
+        return len(schema_result) > 0
+
+    def run_sql(
+        self,
+        sql: str | ClauseElement = "",
+        parameters: dict | None = None,
+        autocommit: bool = False,
+        **kwargs,
+    ) -> CursorResult:
+        """
+        Return the results to running a SQL statement.
+        Whenever possible, this method should be implemented using Airflow Hooks,
+        since this will simplify the integration with Async operators.
+        :param sql: Contains SQL query to be run against database
+        :param parameters: Optional parameters to be used to render the query
+        :param autocommit: specify autocommit setting for the query
+        """
+        if parameters is None:
+            parameters = {}
+
+        if "sql_statement" in kwargs:  # pragma: no cover
+            warnings.warn(
+                "`sql_statement` is deprecated and will be removed in future release"
+                "Please use  `sql` param instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            sql = kwargs.get("sql_statement")  # type: ignore
+
+        # We need to set autocommit=True for specific queries
+        if isinstance(sql, str):
+            if (autocommit is True) or any(
+                query in sql.lower() for query in self._queries_requiring_autocommit
+            ):
+                self.connection.execute(sqlalchemy.text(sql).execution_options(autocommit=True), parameters)
+            else:
+                result = self.connection.execute(
+                    sqlalchemy.text(sql).execution_options(autocommit=autocommit), parameters
+                )
+                return result
+        else:
+            # this is used for append
+            result = self.connection.execute(sql, parameters)
+            return result
+
+    def create_schema_if_needed(self, schema: str | None) -> None:
+        """
+        This function checks if the expected schema exists in the database. If the schema does not exist,
+        it will attempt to create it.
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
```

Comment:
```suggestion
        it will attempt to create it.

        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
```

Reply:
fixed